rockcraft/commands/lifecycle.py

```diff
@@ -18,16 +18,112 @@
 
 import abc
 import textwrap
+from typing import TYPE_CHECKING
 
-from craft_cli import BaseCommand
+from craft_cli import BaseCommand, emit
 from overrides import overrides
 
 from rockcraft import lifecycle
 
+if TYPE_CHECKING:
+    import argparse
+
 
 class _LifecycleCommand(BaseCommand, abc.ABC):
     """Run lifecycle-related commands."""
 
+    @overrides
+    def run(self, parsed_args):
+        """Run the command."""
+        if not self.name:
+            raise RuntimeError("command name not specified")
+
+        emit.trace(f"lifecycle command: {self.name!r}, arguments: {parsed_args!r}")
+        lifecycle.run(self.name, parsed_args)
+
+
+class _LifecycleStepCommand(_LifecycleCommand):
+    """Run lifecycle step commands."""
```

Comment:
nit, but the class does not _Run_, it is the command itself, right?

Reply:
Indeed, removed the "run" word.